src/common/snippets/tests/src/lowered/pass/extracte_loop_invariants.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "lir_test_utils.hpp"
+
+#include "openvino/opsets/opset10.hpp"
+#include "snippets/lowered/pass/extract_loop_invariants.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+using namespace ov::snippets::lowered;
+using namespace ov::snippets::lowered::pass;
+
+class ExtractLoopInvariantsTest : public LoweredPassTestsF {
+public:
+    ExtractLoopInvariantsTest() : LoweredPassTestsF() {
+        comparator.enable(LIRComparator::LIRCmpValues::LOOP_INDICES);
+        comparator.enable(LIRComparator::LIRCmpValues::PORT_DESCRIPTORS);
+        comparator.enable(LIRComparator::LIRCmpValues::PORT_CONNECTORS);
+        comparator.enable(LIRComparator::LIRCmpValues::LOOP_MANAGER);
+    }
+
+    void SetUp() override {
+        pipeline.register_pass<ExtractLoopInvariants>();
+    }
+};
+
+TEST_F(ExtractLoopInvariantsTest, ExtractLoopInvariants) {
```

Comment:
May I ask you to add two tests please?
1. When we extract Expression with all all input ports which are  `LoopPorts`? (as your current test but the second input of `Multiply` is parameter too) - to validate `LoopInfo::update_loop_ports`
2. When all loop is removed after extraction - shape has rank > 1 where last dim is 1 (to validate the Softmax test case with last dim = 1).

Reply:
These two tests are included, thanks Alexandra!

Reply:
Many-many tests were added. Great! Thanks a lot!